packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/DeleteRulesTests.java

```diff
@@ -0,0 +1,182 @@
+    public void deleteRuleInNonExistingRuleSetAndGet404()
+    {
+
+        STEP("Attempt delete the rule in non-existing rule set");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingRuleSet(FAKE_NODE_REF).deleteRule(createdRuleA.getId());
+
+        restClient.assertLastError().statusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Try to delete a non-existing rule and get 404.
+     */
+    @Test(groups = {TestGroup.REST_API, TestGroup.RULES, TestGroup.SANITY})
+    public void deleteNonExistingRuleAndGet404()
+    {
+        STEP("Attempt delete non-existing rule");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet().deleteRule(FAKE_NODE_REF);
+
+        restClient.assertLastError().statusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Check that a user without write permission on folder cannot delete a rule inside it.
+     */
+    public void deleteSingleRuleWithoutPermissionAndGet403()
+    {
+        STEP("Create a user and use them to create a private site containing a folder with a rule");
+        final UserModel privateUser = dataUser.createRandomTestUser();
+        final SiteModel privateSite = dataSite.usingUser(privateUser).createPrivateRandomSite();
+        final FolderModel privateFolder = dataContent.usingUser(privateUser).usingSite(privateSite).createFolder();
+        final RestRuleModel ruleModel = new RestRuleModel();
+        ruleModel.setName("Private site rule");
+        final RestRuleModel createdRule = restClient.authenticateUser(privateUser).withCoreAPI().usingNode(privateFolder).usingDefaultRuleSet().createSingleRule(ruleModel);
+
+        STEP("Try to delete the rule with another user");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(privateFolder).usingDefaultRuleSet().deleteRule(createdRule.getId());
+
+        restClient.assertLastError().statusCodeIs(FORBIDDEN);
+    }
+
+    /**
+     * Check that a user with write permission on folder can delete a rule inside it.
+     */
+    public void deleteSingleRuleWithWritePermissionAndGet204()
+    {
+        STEP("Create a user and use them to create a private site containing a folder with a rule");
+        final UserModel privateUser = dataUser.createRandomTestUser();
+        final SiteModel privateSite = dataSite.usingUser(privateUser).createPrivateRandomSite();
+        final FolderModel privateFolder = dataContent.usingUser(privateUser).usingSite(privateSite).createFolder();
+        final RestRuleModel ruleModel = new RestRuleModel();
+        ruleModel.setName("Private site rule");
+        final RestRuleModel createdRule = restClient.authenticateUser(privateUser).withCoreAPI().usingNode(privateFolder).usingDefaultRuleSet()
+                .createSingleRule(ruleModel);
+
+        STEP("Create a manager in the private site");
+        final UserModel collaborator = dataUser.createRandomTestUser();
+        collaborator.setUserRole(SiteManager);
```

Comment:
I think it's fine to use either a collaborator or a site manager, but you've mixed both terms.